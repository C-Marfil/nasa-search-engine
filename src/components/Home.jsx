import React from "react";
import MovingComponent from "react-moving-text";
import "../styles/home.css";

const Home = () => {
  return (
    <MovingComponent
      className="moving-words"
      type="pulse"
      duration="2000ms"
      delay="1s"
      direction="alternate-reverse"
      timing="ease"
      id="pulse"
      iteration="infinite"
      fillMode="none"
    >
      Welcome, use the Search Bar above to find any pictures held in the public
      NASA archives.
    </MovingComponent>
  );
};

export default Home;
