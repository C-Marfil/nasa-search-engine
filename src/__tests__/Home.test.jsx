import { React, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Home from "../components/Home";

describe("Home", () => {
  it("Snapshot Test", () => {
    const { asFragment } = render(<Home />, { wrapper: Router });

    expect(asFragment()).toMatchSnapshot();
  });

  it("tests that elements are rendered", () => {
    const { getByText } = render(<Home />, { wrapper: Router });
    const movingImage = getByText(
      "Welcome, use the Search Bar above to find any pictures held in the public NASA archives."
    );

    expect(movingImage).toHaveAttribute("type", "pulse");
  });
});
