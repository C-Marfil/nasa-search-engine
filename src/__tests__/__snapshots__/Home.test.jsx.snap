// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="sc-bwzfXH lhcaPW welcome-words"
      direction="alternate-reverse"
      id="pulse"
      type="pulse"
    >
      Welcome, use the Search Bar above to find any pictures held in the public NASA archives.
    </div>
    <hr />
  </div>
</DocumentFragment>
`;
